### PR TITLE
Count work when no result is found

### PR DESCRIPTION
**Pull request by ramiel**
This should fix when grouping and no result is found

### CODE REVIEW

**Comment by B4nan** [comment]
To maintain the full branch coverage, this will either need a test case (that would otherwise fail without that change) or at least ignore hint for istanbul (as using optional chaining usually reports missed branches due to how it is transpiled).

File: packages/knex/src/AbstractSqlDriver.ts [hunk 1]
```diff
@@ -174,7 +174,7 @@ export abstract class AbstractSqlDriver<C extends AbstractSqlConnection = Abstra
       .where(where);
     const res = await this.rethrow(qb.execute('get', false));
 
-    return +res.count;
+    return +res?.count || 0;
```
Comment (B4nan) [inline packages/knex/src/AbstractSqlDriver.ts:177]: Is it even possible that `res` will not be defined? I don't think so...

Comment (ramiel) [reply packages/knex/src/AbstractSqlDriver.ts:177]: this is what happened to me. Sorry I should have filed a bug actually

Comment (ramiel) [reply packages/knex/src/AbstractSqlDriver.ts:177]: I'm running this

```
await this.myRepository.count(
        { externalId: externalId },
        { groupBy: 'user' },
      );
```

this fails if there's no entry in myRepository that satisfies the condition. The error says `cannot read count of undefined`

Comment (B4nan) [reply packages/knex/src/AbstractSqlDriver.ts:177]: ok then add the test case so we have that covered, to one of the `EntityManager` tests, `em.count` should be covered in all of them, so just extend the test that is already there

Comment (B4nan) [reply packages/knex/src/AbstractSqlDriver.ts:177]: we might still need that istanbul ignore hint, as covering lines with optional chaining is buggy

Comment (B4nan) [reply packages/knex/src/AbstractSqlDriver.ts:177]: do you plan to finish this or should I take over?

Comment (ramiel) [reply packages/knex/src/AbstractSqlDriver.ts:177]: Oh sorry, I had no time to look at this. If you can take over would be great

Comment (B4nan) [reply packages/knex/src/AbstractSqlDriver.ts:177]: Cant replicate this. More importantly, using `count` with `groupBy` will pretty much never give you what you want. Grouping will result in multiple rows, but `em.count()` will give you only the first one. It already uses `count distinct` query, what do you need the group by for?

**Pull request closed without merging**